docs/docs-beta/docs/tutorial/pinecone.md

```diff
@@ -0,0 +1,311 @@
+```
+:::
+
+Now that we have a Pinecone account we can incorporate Pinecone into our code. To help make everything consistent, we will create a Pinecone resource similar to the Dagster maintained DuckDB resource we used to ingest the Goodreads data. Creating a custom resource is fairly simple, the only information we need to authenticate with the Pinecone client is their API key: 
+```python
+#resources.py
+class PineconeResource(ConfigurableResource):
+    api_key: str = Field(
+        description=(
+            "Pinecone API key. See https://docs.pinecone.io/reference/api/introduction"
+        )
+    )
+
+    _client: Pinecone = PrivateAttr()
+
+    def setup_for_execution(self, context: InitResourceContext) -> None:
+        self._client = Pinecone(
+            api_key=self.api_key,
+        )
+    ...
+```
+
+For our methods we will want the ability to create an index, retrieve a index so we can upsert records, and the ability to embed inputs. You can see all the details in the repo but this is what the `create_index` method looks like:
+```python
+#resources.py
+    def create_index(
+        self,
+        index_name: str,
+        dimension: int,
+        metric: str,
+        serverless_spec: ServerlessSpec,
+    ):
+        existing_indexes = [
+            index_info["name"] for index_info in self._client.list_indexes()
+        ]
+
+        if index_name not in existing_indexes:
+            self._client.create_index(
+                index_name,
+                dimension=dimension,
+                metric=metric,
+                spec=serverless_spec,
+            )
+            while not self._client.describe_index(index_name).status["ready"]:
+                time.sleep(1)
+```
+
+With our Pinecone resource complete, we can initialize it using the API key that we set the environment variable for:
+```python
+# resources.py
+pinecone_resource = PineconeResource(api_key=dg.EnvVar("PINECONE_API_KEY"))
+```
+
+Using our Pinecone resource it is easy enough to create an asset to manage the index where we will upload our embeddings:
+```python
+# assets.py
+@dg.asset(
+    kinds={"Pinecone"},
+    group_name="processing",
+)
+def pinecone_index(
+    pinecone_resource: PineconeResource,
+):
+    spec = ServerlessSpec(cloud="aws", region="us-east-1")
+    pinecone_resource.create_index(
+        INDEX_NAME,
+        dimension=1024,
+        metric="cosine",
+        serverless_spec=spec,
+    )
+```
+
+### Generate Embeddings
+Between our data and the index, we are ready to create and upload our embeddings. One final asset will bring this all together. We will iterate through DataFrame from `enriched_reviews` and generate embeddings for each review. Then we can include additional metadata that will be the final vectors we upload to Pinecone using our Pinecone resource.
+
+```python
+# assets.py
+@dg.asset(
+    kinds={"Pinecone"},
+    group_name="processing",
+)
+def pinecone_embeddings(
```

Comment:
I think it would be good to return `MaterializeResult` from these assets to include meaningful metadata.